velox/functions/sparksql/tests/MapTest.cpp

```diff
@@ -105,5 +105,89 @@ TEST_F(MapTest, ErrorCases) {
       {inputVector1, inputVector2, inputVector11, inputVector22},
       true);
 }
+
+TEST_F(MapTest, complexTypes) {
+  auto rowKey = makeRowVector({
+      makeFlatVector<int64_t>(1, [](auto row) { return row; }),
+  });
+  auto rowValue = makeRowVector({
+      makeFlatVector<int64_t>(1, [](auto row) { return row; }),
+  });
+  auto mapRowVector = makeMapVector(
+      {
+          0,
+      },
+      rowKey,
+      rowValue);
+  mapSimple("map(c0, c1)", {rowKey, rowValue}, false, mapRowVector);
+
+  auto arrayKey = makeArrayVector<int64_t>({
+      {1, 2, 3, 4, 5},
+  });
+  auto arrayValue = makeArrayVector<int64_t>({
+      {1, 2, 3, 4, 5},
+  });
+  auto arrayMapVector = makeMapVector(
+      {
+          0,
+      },
+      arrayKey,
+      arrayValue);
+  mapSimple("map(c0, c1)", {arrayKey, arrayValue}, false, arrayMapVector);
```

Comment:
All tests seem to be passing only 2 arguments and using same type for K and V. Can you also add tests with more arguments and different types for K and V?

Reply:
Added, thanks.